src/AutoRest.CSharp/MgmtTest/Generation/Mock/MgmtMockTestBaseWriter.cs

```diff
@@ -135,7 +208,7 @@ protected CodeWriterDeclaration WriteGetResource(Resource carrierResource, MockT
         {
             var idVar = new CodeWriterDeclaration($"{carrierResource.Type.Name}Id".ToVariableName());
             _writer.Append($"var {idVar:D} = {carrierResource.Type}.CreateResourceIdentifier(");
-            foreach (var value in testCase.ComposeResourceIdentifierParameterValues(carrierResource.RequestPath))
+            foreach (var value in testCase.ComposeResourceIdentifierParameterValues(carrierResource.RequestPath, generatingSample))
```

Comment:
Same here. If we are meant to reuse this method, this flag should be defined as a virtual property, in the writer that generates mock tests, it returns false, and in the writer that generates samples, it returns true

Reply:
And I did not get the point why when we are writing samples, this method has to return the serialized name of the parameters instead of values? Because the values are placeholders to let the customer fulfill?